packages/jaeger-ui/src/components/TracePage/TraceStatistics/tableValues.tsx

```diff
@@ -43,23 +42,61 @@ function getChildOfSpans(parentID: string, allSpans: Span[]): Span[] {
   return memoizedParentChildOfMap(allSpans)[parentID] || [];
 }
 
-function computeSelfTime(span: Span, allSpans: Span[]): number {
-  if (!span.hasChildren) return span.duration;
-  // We want to represent spans as half-open intervals like [startTime, startTime + duration).
-  // This way the subtraction preserves the right boundaries. However, DRange treats all
-  // intervals as inclusive. For example,
-  //       range(1, 10).subtract(4, 8) => range([1, 3], [9-10])
-  //       length=(3-1)+(10-9)=2+1=3
-  // In other words, we took an interval of length=10-1=9 and subtracted length=8-4=4.
-  // We should've ended up with length 9-4=5, but we got 3.
-  // To work around that, we multiply start/end times by 10 and subtract one from the end.
-  // So instead of [1-10] we get [10-99]. This makes the intervals work like half-open.
-  const spanRange = new DRange(10 * span.startTime, 10 * (span.startTime + span.duration) - 1);
-  const children = getChildOfSpans(span.spanID, allSpans);
-  children.forEach(child => {
-    spanRange.subtract(10 * child.startTime, 10 * (child.startTime + child.duration) - 1);
-  });
-  return Math.round(spanRange.length / 10);
+function computeSelfTime(parentSpan: Span, allSpans: Span[]): number {
+  if (!parentSpan.hasChildren) return parentSpan.duration;
+
+  let parentSpanSelfTime = parentSpan.duration;
+  let previousChildEndTime = parentSpan.startTime;
+
+  const children = getChildOfSpans(parentSpan.spanID, allSpans).sort((a, b) => a.startTime - b.startTime);
+
+  const parentSpanEndTime = parentSpan.startTime + parentSpan.duration;
+
+  for (let index = 0; index < children.length; index++) {
+    const child = children[index];
+
+    const childEndTime = child.startTime + child.duration;
+    const childStartsAfterParentEnded = child.startTime > parentSpanEndTime;
+    const childEndsBeforePreviousChild = childEndTime < previousChildEndTime;
+
+    // parent |..................|
+    // child    |.......|
+    // child     |.....|                      - childEndsBeforePreviousChild is true, skipped
+    // child                         |......| - childStartsAfterParentEnded is true, skipped
+    if (childStartsAfterParentEnded || childEndsBeforePreviousChild) {
+      continue;
+    }
+
+    let nonOverlappingStartTime = child.startTime;
+    let nonOverlappingDuration = child.duration;
+    const childStartsBeforePreviousChildEnds = child.startTime < previousChildEndTime;
+
+    // parent |.....................|
+    // child    |.......|
+    // child        |.....|                  - childStartsBeforePreviousChildEnds is true
+    // child                |.....|          - childStartsBeforePreviousChildEnds is false
```

Comment:
why isn't this case handled (full subtraction of span's time from parent)? The `else` part below only covers when child extends past the parent.

Reply:
This if is for special cases, when child doesn't have any overlap with previous span, then `child.duration` is subtracted from `parentSpanSelfTime`, see `let nonOverlappingDuration = child.duration;`

Code in the newest commit is a bit different.

These links are to the previous commit, without Math.min/max

1. [let nonOverlappingDuration = child.duration;](https://github.com/jaegertracing/jaeger-ui/blob/5935bd1528ed6596cf73abd99fc070ee3265224d/packages/jaeger-ui/src/components/TracePage/TraceStatistics/tableValues.tsx#L71)
1. [parentSpanSelfTime -= nonOverlappingDuration;](https://github.com/jaegertracing/jaeger-ui/blob/5935bd1528ed6596cf73abd99fc070ee3265224d/packages/jaeger-ui/src/components/TracePage/TraceStatistics/tableValues.tsx#L95)